components/cqrs/name.go

```diff
@@ -0,0 +1,42 @@
+package cqrs
+
+import (
+	"fmt"
+	"strings"
+)
+
+// ObjectName name returns object name in format [package].[type name].
+// It ignores if the value is a pointer or not.
+func ObjectName(v interface{}) string {
+	s := fmt.Sprintf("%T", v)
+	s = strings.TrimLeft(s, "*")
+
+	return s
+}
+
+// StructName name returns struct name in format [type name].
+// It ignores if the value is a pointer or not.
+func StructName(v interface{}) string {
+	segments := strings.Split(fmt.Sprintf("%T", v), ".")
+
+	return segments[len(segments)-1]
+}
+
+type namedMessage interface {
+	Name() string
+}
+
+// MessageName returns the name from a message implementing the following interface:
+// 		type namedMessage interface {
+// 			Name() string
+// 		}
+// It ignores if the value is a pointer or not.
+func MessageName(fallback func(v interface{}) string) func(v interface{}) string {
```

Comment:
what do you think about NamedObject?

Reply:
Following the pattern I suggested above, this could be `NamedStruct`, but if we stick with object, `NamedObject` works.

Reply:
:+1: